### ISSUE TITLE
Unable to update defaultRouteDomain for class Tenant

### ISSUE THREAD

**Issue by trinaths** [F5Networks/f5-appsvcs-extension#271]
### Environment
 * Application Services Version: 3.18
 * BIG-IP Version: 14.1.2.2 Build 0.0.7 Point Release 2

### Summary
AS3 is not updating `defaultRouteDomain` changes for a given tenant. But when we delete the partition and resubmit the declaration `defaultRouteDomain` is being updated.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "action": "deploy",
    "class": "AS3",
    "declaration": {
        "class": "ADC",
        "controls": {
            "class": "Controls",
            "userAgent": "CIS/v2.0.0-WIP OCP/v3.11.188"
        },
        "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab915d-c2400352",
        "label": "c2400352",
        "remark": "AS3 Declation for c2400352",
        "schemaVersion": "3.18.0",
        "test_AS3": {
            "Shared": {
                "class": "Application",
                "https_ose_vserver": {
                    "class": "Service_HTTP",
                    "layer4": "tcp",
                    "snat": "auto",
                    "source": "0.0.0.0/0",
                    "translateServerAddress": true,
                    "translateServerPort": true,
                    "virtualAddresses": [
                        "172.16.3.9"
                    ],
                    "virtualPort": 443
                },
                "openshift_default_svc_1": {
                    "class": "Pool",
                    "loadBalancingMode": "round-robin"
                },
                "openshift_insecure_routes": {
                    "class": "Endpoint_Policy",
                    "rules": [
                        {
                            "actions": [
                                {
                                    "event": "request",
                                    "select": {
                                        "pool": {
                                            "use": "openshift_default_svc_1"
                                        }
                                    },
                                    "type": "forward"
                                }
                            ],
                            "conditions": [
                                {
                                    "all": {
                                        "operand": "equals",
                                        "values": [
                                            "foo.com"
                                        ]
                                    },
                                    "event": "request",
                                    "name": "host",
                                    "type": "httpHeader"
                                }
                            ],
                            "name": "osr_default_test"
                        }
                    ],
                    "strategy": "first-match"
                },
                "ose_vserver": {
                    "class": "Service_HTTP",
                    "layer4": "tcp",
                    "policyEndpoint": "/test_AS3/Shared/openshift_insecure_routes",
                    "snat": "auto",
                    "source": "0.0.0.0/0",
                    "translateServerAddress": true,
                    "translateServerPort": true,
                    "virtualAddresses": [
                        "172.16.3.27"
                    ],
                    "virtualPort": 80
                },
                "template": "shared"
            },
            "class": "Tenant",
            "defaultRouteDomain": 10
        }
    },
    "persist": true
}
```

2. Observe the following error response:
```json
        {
            "code": "ERR_REQUEST_FAILED",
            "message": "declaration failed",
            "response": "POST http://admin:XXXXXX@localhost:8100/mgmt/tm/sys/config merge cli script response=400 body={\"code\":400,\"message\":\"Loading configuration process failed.\",\"errorStack\":[],\"apiError\":26214401}",
            "host": "localhost",
            "tenant": "test_AS3",
            "runTime": 1892
        }
```

### Expected Behavior
Update of `defaultRouteDomain` for the tenant must be possible.

### Actual Behavior
Delete the partition and retry the declaration, `defaultRouteDomain` for the tenant is being updated.

**Comment by dstokesf5** [comment]
Thank you for reporting this, it is in our internal product backlog as AUTOTOOL-1368.

**Comment by dstokesf5** [comment]
This was resolved in the 3.21.0 release.
